core/app/c/[communitySlug]/NavLinkSubMenu.tsx

```diff
@@ -0,0 +1,89 @@
+"use client";
+
+import { useEffect, useState } from "react";
+
+import { logger } from "logger";
+import { Collapsible, CollapsibleContent, CollapsibleTrigger } from "ui/collapsible";
+import { useLocalStorage } from "ui/hooks";
+import { ChevronDown } from "ui/icon";
+import {
+	SidebarMenu,
+	SidebarMenuAction,
+	SidebarMenuButton,
+	SidebarMenuItem,
+	SidebarMenuSub,
+	SidebarMenuSubItem,
+	useSidebar,
+} from "ui/sidebar";
+
+export const NavLinkSubMenu = ({
+	icon,
+	text,
+	parentLink,
+	children,
+}: {
+	icon: React.ReactNode;
+	text: string;
+	parentLink?: React.ReactNode;
+	children: React.ReactNode;
+}) => {
+	const [open, persistOpen] = useLocalStorage<boolean>(`nav-link-sub-menu-open-${text}`);
+	const [actuallyOpen, setActuallyOpen] = useState(false);
+	const { setOpen: setSidebarOpen, state: sidebarState } = useSidebar();
+
+	useEffect(() => {
+		let mounted = true;
+		if (mounted) {
+			setActuallyOpen(open ?? false);
+		}
+		return () => {
+			mounted = false;
+		};
+	}, [open]);
```

Comment:
This effect is breaking my brain a bit. Shouldn't `mounted` always be `true` when the effect is called? What does setting it to `false` within the closure do?

Reply:
ugh, sorry that comment took a while, i always accidentally add a "review" and then don't submit it 😅